tests/e2e/pageobjects/cell.page.ts

```diff
@@ -0,0 +1,197 @@
+import { BasePage } from 'wdio-vscode-service'
+import clipboard from 'clipboardy'
+
+const DEFAULT_SEARCH_FOR_CELL_TIMEOUT = 50000
+
+export enum OutputType {
+    TerminalView = 'terminal-view',
+    ShellOutput = 'shell-output',
+    Display = '.display'
+}
+
+export enum StatusBarElements {
+    Copy = 'Copy',
+    Configure = 'Configure',
+    CLI = 'CLI',
+    ShellScript = 'Shell Script',
+    OpenTerminal = 'Open Terminal',
+    BackgroundTask = 'Background Task',
+    StopTask = 'Stop Task'
+}
+
+export interface NotebookCommand {
+    element$: WebdriverIO.Element
+    text: string
+}
+
+export class Cell extends BasePage<{}, { cell: {} }> {
+    public locatorKey = 'cell' as const
+    #cellRow: WebdriverIO.Element
+    #cellText: string
+    constructor(cellRow: WebdriverIO.Element, cellText: string) {
+        // TODO: Write the locator map
+        super({ cell: {} })
+        this.#cellRow = cellRow
+        this.#cellText = cellText
+    }
+
+    async #getCurrentExecutionRow(expectedContent?: string) {
+        const successRows = await browser.$$('.codicon-notebook-state-success')
+        for await (const row of successRows) {
+            const executionRow = await row
+                .parentElement()
+                .parentElement()
+                .parentElement()
+                .parentElement()
+                .parentElement()
+            const cellEditor = await executionRow.$('.cell-editor-container')
+            const text = await cellEditor.getText()
+            if (text === (expectedContent ?? this.#cellText)) {
+                return executionRow
+            }
+        }
+    }
+
+    async focus() {
+        const container = await this.#cellRow.parentElement().parentElement()
+        await container.click()
+    }
+
+    async run() {
+        const container = await this.#cellRow.parentElement().parentElement()
+        await container.$('.run-button-container').click()
+    }
+
+    async #getStatusBarElements(): Promise<NotebookCommand[]> {
+        const executionRow = await this.#getCurrentExecutionRow()
+        let commands: NotebookCommand[] = []
```

Comment:
we never re-assign to this variable so we can make it a const
```suggestion
        const commands: NotebookCommand[] = []
```